protos/flyteidl/service/dataproxy.proto

```diff
@@ -63,6 +64,51 @@ message CreateDownloadLocationResponse {
   google.protobuf.Timestamp expires_at = 2;
 }
 
+enum ArtifactType {
+  ARTIFACT_TYPE_UNDEFINED = 0;
+  ARTIFACT_TYPE_DECK = 1;
```

Comment:
Do we also need to  define ARTIFACT_TYPE_SOURCE which would be used for `pyflyte run --remote` and flytectl fast registration

Reply:
That would be used by CreateUploadRequest. Let me know what you think @EngHabu